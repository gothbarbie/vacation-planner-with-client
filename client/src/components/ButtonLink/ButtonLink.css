input[type="submit"],
input[type="reset"],
input[type="button"],
button {
    background: none;
    border: 0;
    color: inherit;
    /* cursor: default; */
    font: inherit;
    line-height: normal;
    overflow: visible;
    padding: 0;
    -webkit-appearance: button; /* for input */
    -webkit-user-select: none; /* for button */
       -moz-user-select: none;
        -ms-user-select: none;
}
input::-moz-focus-inner,
button::-moz-focus-inner {
    border: 0;
    padding: 0;
}

/* Make `a` like a button */
[role="button"] {
    color: inherit;
    cursor: default;
    display: inline-block;
    text-align: center;
    text-decoration: none;
    white-space: pre;
    -webkit-user-select: none;
       -moz-user-select: none;
        -ms-user-select: none;
}


[role="button"],
input[type="submit"],
input[type="reset"],
input[type="button"],
button,
.btn {
  font-size: 24px;
  font-family: 'Avenir Next', sans-serif;
  font-weight: 100;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 42px;
  min-width: 138px;
  cursor: pointer;
  padding: 0 1rem;
}

.btn:link, .btn:visited {
  text-decoration: none;
}
.btn:hover, .btn:active {
  text-decoration: none;
}

.btn__default {
  background: #ccc;
  color: #fff;
}

.btn__primary {
  background: linear-gradient(180deg, #369ED9 0%, #308BBF 100%);
}

.btn__secondary {
  background: linear-gradient(180deg, #EB5757 0%, #CC4C4C 100%);
}

.btn--disabled {
  background: linear-gradient(180deg, #CED5D9 0%, #B6BCBF 100%);
  color: rgba(51, 51, 51, 0.5);
  cursor: not-allowed;
}
