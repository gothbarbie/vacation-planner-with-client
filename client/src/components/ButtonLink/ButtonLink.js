// @flow

import * as React from 'react'
import classnames from 'classnames'

import './ButtonLink.css'

type Props = {
  children: React.ChildrenArray<any>,
  primary?: boolean,
  secondary?: boolean,
  disabled?: boolean,
  url?: string,
}

const ButtonLink = ({ children, primary, secondary, disabled, url }: Props) => (
  <a
    className={classnames('btn btn__default', {
      btn__primary: primary,
      btn__secondary: secondary,
      'btn--disabled': disabled,
    })}
    disabled={disabled}
    href={url}>
    {children}
  </a>
)

export default ButtonLink
