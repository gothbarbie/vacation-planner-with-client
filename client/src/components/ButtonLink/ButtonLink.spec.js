import React from 'react'
import { shallow } from 'enzyme'

import ButtonLink from './ButtonLink'

describe('components/Input', () => {
  const buttonLink = shallow(<ButtonLink />)

  it('renders', () => {
    expect(buttonLink).toMatchSnapshot()
  })
})
