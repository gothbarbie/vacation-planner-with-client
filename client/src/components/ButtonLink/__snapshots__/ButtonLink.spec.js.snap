// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input renders 1`] = `
<a
  className="btn btn__default"
/>
`;
